src/lwc/gePaymentGatewayManagement/gePaymentGatewayManagement.js

```diff
@@ -0,0 +1,148 @@
+ *     COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *     INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *     BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *     LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *     CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *     LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *     ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *     POSSIBILITY OF SUCH DAMAGE.
+ */
+import { LightningElement } from 'lwc';
+import { buildErrorMessage } from 'c/utilTemplateBuilder';
+import { isEmpty } from 'c/utilCommon';
+
+import messageLoading from '@salesforce/label/c.labelMessageLoading';
+
+import setGatewayId from '@salesforce/apex/PS_GatewayManagement.setGatewayId';
+import getGatewayIdFromConfig from '@salesforce/apex/PS_GatewayManagement.getGatewayIdFromConfig';
+import checkForElevateCustomer from '@salesforce/apex/PS_GatewayManagement.isElevateCustomer';
+
+export default class GePaymentGatewayManagement extends LightningElement {
+
+    showSpinner;
+    gatewayId;
+
+    isReadOnly = true;
+    isElevateCustomer;
+    errorMessage;
+
+    CUSTOM_LABELS = { messageLoading };
+
+    connectedCallback() {
+        checkForElevateCustomer()
+            .then(isElevateCustomer => {
+                this.isElevateCustomer = !!isElevateCustomer;
+            })
+            .catch(ex => {
+                this.errorMessage = buildErrorMessage(ex);
+                this.isError = true;
+            });
+
+        this.getGatewayId();
+    }
+
+    _isSuccess;
+    get isSuccess() {
+        return this._isSuccess;
+    }
+    set isSuccess(value) {
+        this._isSuccess = value;
+
+        if (value) { this.isError = false; }
+    }
+
+    _isError;
+    get isError() {
+        return this._isError;
+    }
+    set isError(value) {
+        this._isError = value;
+
+        if (value) { this.isSuccess = false; }
+    }
+
+    handleEdit() {
+        this.isReadOnly = false;
+    }
+
+    handleCancel() {
+        this.isReadOnly = true;
+    }
+
+    async handleSave(event) {
+        this.resetAlert();
+
+        let gatewayId = this.validateGatewayId();
+
+        if (isEmpty(gatewayId)) {
+            return;
+        }
+
+        this.showSpinner = true;
+
+        try {
+
+            await setGatewayId({ gatewayId: gatewayId});
+
+            this.isReadOnly = true;
+            this.isSuccess = true;
+            this.showSpinner = false;
+        } catch(ex) {
+            this.errorMessage = buildErrorMessage(ex);
+            this.showSpinner = false;
+            this.isError = true;
+        }
+    }
+
+    async getGatewayId() {
+        try {
+            this.gatewayId = await getGatewayIdFromConfig();
+        } catch(ex) {
+            this.errorMessage = buildErrorMessage(ex);
+            this.isError = true;
+        }
```

Comment:
If there is not a specific exception here that is expected to occur then I think this try catch block should be removed.

Reply:
Good call. done!